src/test/fuzz/base_encode_decode.cpp

```diff
@@ -10,45 +10,86 @@
 #include <util/string.h>
 
 #include <cassert>
-#include <cstdint>
 #include <string>
 #include <vector>
 
 using util::TrimString;
-using util::TrimStringView;
 
-FUZZ_TARGET(base_encode_decode)
+FUZZ_TARGET(base58_encode_decode)
 {
-    const std::string random_encoded_string(buffer.begin(), buffer.end());
+    std::string random_string(buffer.begin(), buffer.end());
+    std::vector<unsigned char> random_data{ToByteVector(random_string)};
 
+    // Decode/Encode roundtrip
     std::vector<unsigned char> decoded;
-    if (DecodeBase58(random_encoded_string, decoded, 100)) {
-        const std::string encoded_string = EncodeBase58(decoded);
-        assert(encoded_string == TrimStringView(encoded_string));
-        assert(ToLower(encoded_string) == ToLower(TrimString(random_encoded_string)));
+    if (DecodeBase58(random_string, decoded, 100)) {
+        auto encoded_string{EncodeBase58(decoded)};
+        assert(encoded_string == TrimString(random_string));
+        assert(encoded_string.empty() || !DecodeBase58(encoded_string, decoded, decoded.size() - 1));
```

Comment:
Could test slightly more aggressively in line with the removed test:
https://github.com/bitcoin/bitcoin/blob/a5b4e42a575b4680a9eedb6ab2275d2a723cc20d/src/test/base58_tests.cpp#L93-L94

```suggestion
        assert(encoded_string.empty() || !DecodeBase58(encoded_string, decoded, fuzzed_data_provider.ConsumeIntegralInRange(0, decoded.size() - 1)));
```

Reply:
Good idea, this will simplify the string generation as well since it requires a `FuzzedDataProvider` anyway